src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -56,7 +68,14 @@ public override T[] Rent(int minimumLength)
 
             // Gettings here means we have too big of a request OR all the buckets from 
             // index through _buckets.Length are taken so we need to allocate a buffer on-demand.
-            return new T[Utilities.GetMaxSizeForBucket(index)];
+            int maxLength = Utilities.GetMaxSizeForBucket(_buckets.Length);
+            buffer = new T[Utilities.GetMaxSizeForBucket(index)];
+            if (buffer.Length > maxLength) 
+                _eventSource.BufferOverMaximumAllocated(buffer.GetHashCode(), buffer.Length, maxLength);
+            else 
+                _eventSource.BufferAllocated(buffer.GetHashCode(), buffer.Length);
```

Comment:
Another example where I think we want to first check whether logging is enabled, e.g.
```C#
var buffer = new T[Utilities.GetMaxSizeForBucket(index)];
var log = ArrayPoolEventSource.Log;
if (log.IsEnabled())
{
    int maxLength = Utilities.GetMaxSizeForBucket(_buckets.Length);
    if (buffer.Length > maxLength)
        log.BufferOverMaximumAllocated(...);
    else
        log.BufferAllocated(...);
}
return buffer;
```